manim/mobject/text/tex_mobject.py

```diff
@@ -519,9 +521,11 @@ def __init__(
     def fade_all_but(self, index_or_string: int | str, opacity: float = 0.5) -> None:
         arg = index_or_string
         if isinstance(arg, str):
-            part = self.get_part_by_tex(arg)
+            part: VGroup | VMobject | None = self.get_part_by_tex(arg)
+            if part is None:
+                raise Exception("Could not locate part by provided tex string")
```

Comment:
I like this addition! Could you please also include the given TeX string in the error?

Would it make sense that this was a `KeyError`? `arg` is sort of a key to look for in the TeX, but this is not really a mapping...
```suggestion
                raise Exception(f"Could not locate part by provided tex string '{arg}'.")
```